src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -84,7 +86,13 @@ private DelegatedExecution getDelegatedExecutionForFieldDefinition(GraphQLFieldD
                                                                 List<MergedField> mergedFields,
                                                                 DelegatedExecution delegatedExecution,
                                                                 ExecutionStepInfo executionStepInfo) {
-        Document query = mergedFieldsToDocument.mergedSelectionSetToDocument(mergedFields);
+        SourceQueryTransformer queryTransformer = new SourceQueryTransformer(context);
```

Comment:
please make it an instance variable and pass along the `executionContext`, see the other classes.

Reply:
I cannot because this is per execution (and it is stateful). The idea here is that you can add multiple operations (query and / or mutation). I can create something like Provider class but this would be just unnecessary for now. It can be done when needed.

Reply:
👍